airgun/entities/host_new.py

```diff
@@ -272,15 +435,146 @@ def get_parameters(self, entity_name):
+        self,
+        entity_name,
+        parameter_to_change,
+        new_parameter_name=None,
+        new_parameter_type=None,
+        new_parameter_value=None,
+    ):
+        """
+        Function that can edit parameter name, type or value separetely or all of them at once.
+
+        Args:
+            entity_name: Name of the entity to be edited
+            parameter_to_change: Name of the parameter to be edited
+            new_parameter_name: New name of the parameter
+            new_parameter_type: New type of the parameter
+                [string, boolean, integer, real, array, hash, yaml, json]
+            new_parameter_value: New value of the parameter
+
+        Raises:
+            ValueError: No new parameter name, type or value provided
+            ValueError: If parameter type is not valid
+            ValueError: If parameter_to_change is not found on host
+            ValueError: If parameter with new name already exists
+        """
+
+        if not any([new_parameter_name, new_parameter_type, new_parameter_value]):
+            raise ValueError(
+                'At least one of the new_parameter_name, new_parameter_type, new_parameter_value '
+                'must be provided!'
+            )
+        # Sanitize input
+        new_parameter_type = new_parameter_type.lower()
+        available_types = ['string', 'boolean', 'integer', 'real', 'array', 'hash', 'yaml', 'json']
+        if new_parameter_type not in available_types:
+            raise ValueError(f'Parameter type {new_parameter_type} is not valid!')
+
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+        view.parameters.click()
+        view.parameters.searchbar.fill(parameter_to_change)
+        view.wait_displayed()
+        if view.parameters.parameters_table.row_count == 0:
+            raise ValueError(
+                f'Parameter {parameter_to_change} not found on {entity_name}, '
+                'thus cannot be edited.'
+            )
+        elif view.parameters.parameters_table[0][0].text != parameter_to_change:
+            raise ValueError(
+                f'Parameter {parameter_to_change} not found on {entity_name}, '
+                'thus cannot be edited.'
+            )
+
+        view.parameters.searchbar.fill(new_parameter_name)
+        view.wait_displayed()
+        if view.parameters.parameters_table.row_count != 0:
+            if view.parameters.parameters_table[0][0].text == new_parameter_name:
+                raise ValueError(
+                    f'Cannot rename {parameter_to_change} to {new_parameter_name}. '
+                    'This parameter already exists.'
+                )
+
+        view.parameters.searchbar.fill(parameter_to_change)
+        view.wait_displayed()
+        view.parameters.parameters_table[0][4].widget.click()
+        view.wait_displayed()
+        if new_parameter_name:
```

Comment:
`fill` for the whole form?

Reply:
Similar explanation as here: https://github.com/SatelliteQE/airgun/pull/833#discussion_r1179050031 
It is not implemented as form... and widgetastic does not support forms.